library/kani_macros/src/lib.rs

```diff
@@ -89,17 +89,67 @@ pub fn derive_arbitrary(item: TokenStream) -> TokenStream {
     derive::expand_derive_arbitrary(item)
 }
 
+/// Add a precondition to this function.
+///
+/// This is part of the function contract API, together with [`ensures`].
+///
+/// The contents of the attribute is a condtition over the input values to the
```

Comment:
Well for (safe rust) constants it shouldn't matter since they can't change so they either satisfy the conditions always or never.

Static's yes, but I haven't investigated yet if they work with CBMC yet and I'm focusing on safe rust with wouldn't allow that anyway.